### PR TITLE
feat: add shadow support to aria-required-children

### PR DESCRIPTION
**Pull request by marcysutton**
I didn't add any shadow DOM tests for `aria-owns` because ID attributes get reset inside of shadow boundaries, so that technique won't work. Are there any other Shadow DOM scenarios I should consider?

Closes https://github.com/dequelabs/axe-core/issues/421

### CODE REVIEW

File: test/checks/aria/required-children.js [hunk 1]
```diff
@@ -10,97 +11,141 @@ describe('aria-required-children', function () {
 		}
 	};
 
+	function checkSetup (html, options, target) {
```
Comment (dylanb) [inline test/checks/aria/required-children.js]: This will have to be refactored once @WilcoFiers pull request gets merged - perhaps you should copy that change so that it will merge and eliminate refactoring?

File: lib/checks/aria/required-children.js [hunk 2]
```diff
@@ -13,17 +13,17 @@ function owns(node, role, ariaOwned) {
...
+		!!axe.utils.querySelectorAll(virtualTree, selector)[0];
 }
 
 function ariaOwns(nodes, role) {
 	var index, length;
 
 	for (index = 0, length = nodes.length; index < length; index++) {
 		if (nodes[index] === null) { continue; }
-		if (owns(nodes[index], role, true)) {
+		let virtualTree = axe.utils.getFlattenedTree(nodes[index]);
```
Comment (dylanb) [inline lib/checks/aria/required-children.js]: I think you should use `axe.utils.getNodeFromTree()` instead of recreating the flattened tree

Comment (marcysutton) [inline lib/checks/aria/required-children.js]: That's the whole problem though–I don't have access to the virtualNode at that time, and the arguments for `getNodeFromTree` include `vNode, node`. This method is called recursively on a collection of elements from `idRefs`, which don't return virtualNodes. There's a comment in that function about only working with the light DOM. How would I get the virtualNode in this `ariaOwns` function without getting the flattened tree first?

File: test/checks/aria/required-children.js (hunk 1 above)
Comment (marcysutton) [inline test/checks/aria/required-children.js]: Ok, I've merged in that branch and used the checkSetup in testutils.

**Comment by dylanb** [comment]
NOTE: we need to wait for the other PR to be merged before finalizing review on this PR

**Comment by marcysutton** [comment]
@dylanb did you mean to wait for the accessible-text PR? I rebased to get rid of conflicts with that branch and it should be good to go now.

**Comment by dylanb** [comment]
@WilcoFiers the downside to changing to ES6 notation is that we lose the history of why a change was made, so I am also completely fine with leaving this as-is
